frontend/src/api/RestaurantApi.tsx

```diff
@@ -0,0 +1,43 @@
+import { SearchState } from '@/pages/SearchPage';
+import { RestaurantResponse } from '@/types';
+import { useAuth0 } from '@auth0/auth0-react';
+import { useQuery } from 'react-query';
+
+const API_BASE_URL = import.meta.env.VITE_API_BASE_URL;
+
+export const useSearchRestaurants = (
+  searchState: SearchState,
+  city: string
+) => {
+  const { getAccessTokenSilently } = useAuth0();
+
+  const createSearchRestaurant = async (): Promise<RestaurantResponse> => {
+    const accessToken = await getAccessTokenSilently();
+    const params = new URLSearchParams();
+    params.set('searchQuery', searchState.searchQuery);
+
+    const response = await fetch(
+      `${API_BASE_URL}/api/restaurant/search/${encodeURIComponent(city)}?${params.toString()}`,
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Food-Ordering-App/pull/43#issuecomment-2408154714>#86982f</a><span> - 10/11/2024, 09:46 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The API endpoint construction might be vulnerable to path traversal attacks if the 'city' parameter is not properly sanitized.
</div>
<div id="fix">
<b>Fix:</b> Implement proper input validation and sanitization for the 'city' parameter before using it in the API endpoint URL. Use a whitelist approach to allow only valid characters and patterns for city names.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -14,9 +14,15 @@
    const createSearchRestaurant = async (): Promise<RestaurantResponse> => {
      const accessToken = await getAccessTokenSilently();
      const params = new URLSearchParams();
      params.set('searchQuery', searchState.searchQuery);
 
 +    const sanitizedCity = sanitizeCity(city);
 +    if (!sanitizedCity) {
 +      throw new Error('Invalid city name');
 +    }
 +
      const response = await fetch(
 -      `${API_BASE_URL}/api/restaurant/search/${encodeURIComponent(city)}?${params.toString()}`,
 +      `${API_BASE_URL}/api/restaurant/search/${sanitizedCity}?${params.toString()}`,
        {
          method: 'GET',
          headers: {
 @@ -43,3 +49,11 @@
 
    return { results, isLoading };
  };
 +
 +function sanitizeCity(city: string): string | null {
 +  // Implement a whitelist approach, allowing only alphanumeric characters, spaces, and hyphens
 +  const sanitized = city.replace(/[^a-zA-Z0-9\s-]/g, '');
 +  // Ensure the sanitized city is not empty and meets a minimum length requirement
 +  return sanitized.length >= 2 ? sanitized : null;
 +}

```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged